core/controllers/reader.py

```diff
@@ -329,7 +329,8 @@ def post(self, unused_exploration_id):
         arguments.
         """
         # A domain object representing the old state.
-        old_state = exp_domain.State.from_dict(self.payload.get('old_state'))
+        old_state = exp_domain.State.from_dict(self.payload.get(
+            'old_state_dict'))
```

Comment:
Why are you changing the name of the field that's coming in via the payload? I expect this to break stuff...